### PR TITLE
Autopopulate CopySourceSSECustomerKeyMD5 for s3

### PR DESCRIPTION
**Pull request by kyleknap**
This is implementing logic used for this PR: https://github.com/aws/aws-cli/pull/1487. We have it for SSECustomer members but not the copy source one which may be needed when calling CopyObject.

cc @jamesls @mtdowling @rayluo @JordonPhillips

### CODE REVIEW

**Comment by JordonPhillips** [comment]
Looks good to me :ship:

File: botocore/handlers.py [hunk 1]
```diff
@@ -170,21 +170,37 @@ def sse_md5(params, **kwargs):
...
-    params['SSECustomerKeyMD5'] = key_md5_str
+    params[sse_key_member] = key_b64_encoded
+    params[sse_md5_member] = key_md5_str
 
 
-def _needs_s3_sse_customization(params):
-    return (params.get('SSECustomerKey') is not None and
-            'SSECustomerKeyMD5' not in params)
+def _needs_s3_sse_customization(params, sse_member_prefix='SSECustomer'):
+    return (params.get(sse_member_prefix+'Key') is not None and
```
Comment (mtdowling) [inline botocore/handlers.py]: I think we typically include spaces between operators

**Comment by mtdowling** [comment]
:ship:

File: botocore/handlers.py [hunk 2]
```diff
@@ -170,21 +170,37 @@ def sse_md5(params, **kwargs):
...
-    params['SSECustomerKey'] = key_b64_encoded
-    params['SSECustomerKeyMD5'] = key_md5_str
+    params[sse_key_member] = key_b64_encoded
+    params[sse_md5_member] = key_md5_str
 
 
-def _needs_s3_sse_customization(params):
-    return (params.get('SSECustomerKey') is not None and
-            'SSECustomerKeyMD5' not in params)
+def _needs_s3_sse_customization(params, sse_member_prefix='SSECustomer'):
```
Comment (jamesls) [inline botocore/handlers.py]: Any reason this needs to have a default value if we're always providing it when we call `_needs_s3_sse_customization`?

File: botocore/handlers.py (hunk 1 above)
Comment (kyleknap) [inline botocore/handlers.py]: hmm flake8 did not catch that. I am changing it, but for future references, there are some nuances to that rule in pep8 if you have multiple operators on the same line: https://www.python.org/dev/peps/pep-0008/#other-recommendations.

File: botocore/handlers.py (hunk 2 above)
Comment (kyleknap) [inline botocore/handlers.py]: Nah we do not need a default. Just forgot to remove it upon implementation.

**Comment by kyleknap** [comment]
@jamesls I updated. Let me know what you think.

**Comment by jamesls** [comment]
:shipit:

**Pull request merged**